### PR TITLE
Replace NixOS/nixpkgs-channels with NixOS/nixpkgs

### PR DESCRIPTION
**Pull request by srhb**
Fixes https://github.com/nmattia/niv/issues/168

This is a completely naive search-and-replace with some artisanal description updating sprinkled on top. nixpkgs mirrors all the way down to 18.09, so I _think_ this is sufficient(?)

I've run the tests via `repl > :main` -- is there anything else I should do to verify?

### CODE REVIEW

**Comment by srhb** [comment]
Hmm, okay, guess I broke something. I'll try and figure it out.

**Comment by srhb** [comment]
`./script/test` runs for me with "all good". I don't have a Mac, so if I _did_ break something, I'm not sure how to verify it. :)

**Comment by domenkozar** [comment]
@nmattia this one would be great to merge as channels confuse newcomers :)

**Comment by Sir4ur0n** [comment]
Hi,

Can this PR be reconsidered for review please? I just started using Niv and indeed, I got quite confused on this :sweat_smile: I can only imagine many (new) users feel the same.

Note: let me know if I can help, though I think this PR is pretty much done?

**Comment by nmattia** [comment]
Sorry for the delay, I was out for a while. I'll have a look at this asap!

**Comment by nmattia** [comment]
Alright, I cleaned up a bit:

* https://github.com/nmattia/niv/pull/237 upgraded the nixpkgs used to build niv to nixpkgs@20.03; that means you don't have to modify the nix/sources.json.
* https://github.com/nmattia/niv/pull/239 cleaned up a bit how niv deals with importing nixpkgs; this means you can now simply change `defaultNixpkgsRepo` and you should be good to go. The tests can still use nixpkgs-channels, no need to change that anymore either.

I guess the readme still need to be updated though! Thanks for doing this!

**Comment by srhb** [comment]
I can drop those changes, but I'm not sure there's any point if you want to keep using the deprecated repository regardless (but why? :smile:)

To clarify: NixOS/nixpkgs-channels is deprecated. There's no date for it going away, but the proper repository to get all branches from is now NixOS/nixpkgs.

Furthermore I'm a bit surprised you've baked in the `release-$foo` branches in your above PRs. `release-$foo` branches are the untested, pre-Hydra CI branches -- like "master" but for a stable branch instead of for `{nixos,nixpkgs}-unstable"` whereas `nixpkgs-$foo` and `nixos-$foo` are post-Hydra CI branches, that correspond to the deprecated nixpkgs-channels branches. Was that intentional?

**Comment by nmattia** [comment]
> I can drop those changes

Please don't! What I mean is that you now need way fewer changes. In particular, you don't need to update the the tests. Sorry for the confusion!

> NixOS/nixpkgs-channels is deprecated.

Yep, and I've never liked it. Let's nuke it.


> surprised you've baked in the release-$foo branches in your above PRs.  [...] Was that intentional?

No! I always thought `nixos-FOO` was the nixos release and `release-FOO` was the nixpkgs release. Thanks for clarifying that!

**Comment by srhb** [comment]
Ah, ok, thank you for clarifying! :smile:  I'll clean this up tonight then. :)

**Comment by srhb** [comment]
Done. I didn't touch tests, nor change the "release" branches, because I'm not sure what you want to point them to.
I added the default user/repo to the options parser default values, because I think they were just missed.

File: src/Niv/Cli.hs [hunk 1]
```diff
@@ -139,7 +139,7 @@ parseCmdInit = Opts.info (cmdInit <$> parseNixpkgs <**> Opts.helper) $ mconcat d
         Opts.showDefault <>
         Opts.help "Use a custom nixpkgs repository from GitHub." <>
         Opts.metavar "OWNER/REPO" <>
-        Opts.value (Nixpkgs "NixOS" "nixpkgs-channels")
+        Opts.value (Nixpkgs defaultNixpkgsUser defaultNixpkgsRepo)
```
Comment (nmattia) [inline src/Niv/Cli.hs:142]: oh good catch!

**Comment by nmattia** [comment]
I'll release a new version tomorrow!

**Pull request merged**